compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml

```diff
@@ -243,6 +243,26 @@ data AnyContractKey = AnyContractKey
   , getanyContractKeyTemplateRep : TemplateTypeRep
   }
 
+#ifdef DAML_GENMAP
+-- We do not have a general Eq instance for Any since Any is not exposed
+-- to users directly and that instance will crash on some values but
+-- we can have total Eq instances for AnyTemplate, AnyChoice and AnyContractKey
+-- since those are guaranteed to be serializable.
+eqAny : Any -> Any -> Bool
+eqAny = primitive @"BEEqual"
+
+instance Eq AnyTemplate where
+  AnyTemplate a == AnyTemplate b = eqAny a b
+
+instance Eq AnyChoice where
+  AnyChoice aAny aRep == AnyChoice bAny bRep =
+    eqAny aAny bAny && aRep == bRep
```

Comment:
```suggestion
    aRep == bRep && eqAny aAny bAny
```
Somehow comparing the type reps after the values _feels_ wrong to me. Same below.

Reply:
I disagree, this is not the type of the choice but the type of the template the choice belongs to. Having the same choice type on two different templates is an edge case so it make sense to test for that only if the other test isn’t conclusive.